src/dev/impl/DevToys/Models/NumberBaseDictionary.cs

```diff
@@ -0,0 +1,68 @@
+﻿using System;
+using System.Collections.Generic;
+using System.Linq;
+using System.Text;
+using System.Threading.Tasks;
+
+namespace DevToys.Models
+{
+    public class NumberBaseDictionary : IEquatable<NumberBaseDictionary>
+    {
+        public static readonly NumberBaseDictionary Base16Dictionary = new(
+            _dictionary: "0123456789ABCDEF".ToCharArray(), 
+            formatting: true);
+
+        public static readonly NumberBaseDictionary RFC4648Base32Dictionary = new(
+            _dictionary: "ABCDEFGHIJKLMNOPQRSTUVWXYZ234567".ToCharArray(),
+            formatting: false);
+
+        public static readonly NumberBaseDictionary RFC4648Base32ExHexDictionary = new(
+            _dictionary: "0123456789ABCDEFGHIJKLMNOPQRSTUV".ToCharArray(),
+            formatting: false);
+
+        public static readonly NumberBaseDictionary RFC4648Base64Dictionary = new(
+            _dictionary: "ABCDEFGHIJKLMNOPQRSTUVWXYZabcdefghijklmnopqrstuvwxyz0123456789+/".ToCharArray(),
+            formatting: false);
+
+        public static readonly NumberBaseDictionary RFC4648Base64UrlEncodeDictionary = new(
+             _dictionary: "ABCDEFGHIJKLMNOPQRSTUVWXYZabcdefghijklmnopqrstuvwxyz0123456789-_".ToCharArray(),
+             formatting: false);
+        public static NumberBaseDictionary DefaultDictionary { get; } = RFC4648Base64Dictionary;
+
+        public char[] Dictionary { get; }
```

Comment:
I'd suggest using a `string` instead. You can still do `return Dictionary[index];` on line 50. Using a string would avoid us to allocate a string and then a char array right after.